table.cc

```diff
@@ -145,9 +145,12 @@ table::make_reader(schema_ptr s,
                            const io_priority_class& pc,
                            tracing::trace_state_ptr trace_state,
                            streamed_mutation::forwarding fwd,
-                           mutation_reader::forwarding fwd_mr) const {
+                           mutation_reader::forwarding fwd_mr,
+                           std::optional<query::max_result_size> max_memory_for_unlimited_query) const {
```

Comment:
ps. this (and any other parameter that is the same for a reader graph)  is a good candidate for moving into the permit.